dockerfile_parse/parser.py

```diff
@@ -767,7 +767,12 @@ def add_lines(self, *lines, **kwargs):
         for stage in range(len(froms)-2, -1, -1):  # e.g. 0 for single or 2, 1, 0 for 3 stages
             start, finish = froms[stage], froms[stage+1]
             linenum = start['endline'] + 1 if at_start else finish['startline']
-            if skip_scratch and froms[stage]['value'] == 'scratch':
+
+            from_line_value = froms[stage].get('value')
+            image = None
+            if from_line_value:
+                image, _ = image_from(from_line_value)
```

Comment:
Could be just
```python
image, _ = image_from(froms[stage].get('value') or '')
```
Is even possible that `value` is not defined?

Reply:
yes we have that case in unit tests